### PR TITLE
[todo] add ++close for :terminal

### PR DESCRIPTION
**Pull request by mattn**
also add `{"term_autoclose": 1}` for term_start.

### CODE REVIEW

**Comment by mattn** [comment]
I'll add test.

File: runtime/doc/terminal.txt [hunk 1]
```diff
@@ -89,6 +89,9 @@ Syntax ~
 			parenthesis.  E.g. if "gdb" exists the second terminal
 			buffer will use "!gdb (1)".
 
+			If {++autoclose} is provided, the terminal close
```
Comment (dmerejkowsky) [inline runtime/doc/terminal.txt]: closeS (or will close)

File: runtime/doc/terminal.txt [hunk 2]
```diff
@@ -89,6 +89,9 @@ Syntax ~
...
+			automatically when the job terminated.
+
 When the buffer associated with the terminal is wiped out the job is killed,
 similar to calling `job_stop(job, "kill")`
```
Comment (dmerejkowsky) [inline runtime/doc/terminal.txt]: when the job terminates

File: runtime/doc/eval.txt [hunk 3]
```diff
@@ -8037,8 +8037,9 @@ term_start({cmd}, {options})				*term_start()*
 		terminal then the callback function for that part is not used.
 
 		There is one extra option:
```
Comment (k-takata) [inline runtime/doc/eval.txt]: There are two extra options:

**Comment by brammool** [comment]
Yasuhiro Matsumoto wrote:

> also add `{"term_autoclose": 1}` for term_start.

Thanks for working on this.

I have been wondering about the syntax of options like autoclose.
We already have two existing mechanisms we could use:

	:term <autoclose> {command}
	:term ++autoclose {command}

We could even do:
	:term +autoclose {command}
But that would be a new one, and confusing with "+cmd" for some
commands.  Also using "-opt" or "--opt", like used for shell commands,
would introduce a new mechanism, and probably can't be used with some
commands.

The two mechanisms were done for various reasons for different commands,
here we have a new command, thus we have a choice.  Hopefully we can use
the same for other commands later.

I don't expect any {command} to start with "<" or "++", so both could be
used.

I have a slight preference for using ++autoclose, mainly because :edit
uses ++opt and that's the most common command.

-- 
Shaw's Principle: Build a system that even a fool can use, and
only a fool will want to use it.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mattn** [comment]
Vim have `:e ++enc=latin-1` or `:e ++ff=unix` or `:e ++bin`. Also `:command -nargs=1`. `:map <buffer>` as you mentioned. My opinion is `<xxx>` is provided for map command.

**Comment by mattn** [comment]
I don't have strong opinion.  So I'll change this to use `<autoclose>`.

**Comment by mattn** [comment]
Sorry, I don't really understand which style do you expect. When you decide, I'll update this patch. :)

**Comment by brammool** [comment]
Yasuhiro wrote:

> Sorry, I don't really understand which style do you expect. When you
> decide, I'll update this patch. :)

So far I prefer ++close.  Mainly because it's easier to type and also
because the ++ arguments in :edit are probably the most often used ones.

-- 
BEDEVERE: Look!  It's the old man from scene 24 - what's he Doing here?
ARTHUR:   He is the keeper of the Bridge.  He asks each traveler five
          questions ...
GALAHAD:  Three questions.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mattn** [comment]
Just remove `auto` for `:terminal` and keep `term_autoclose` for term-options.

File: runtime/doc/eval.txt [hunk 4]
```diff
@@ -8036,7 +8036,7 @@ term_start({cmd}, {options})				*term_start()*
 		connected to the terminal.  When I/O is connected to the
 		terminal then the callback function for that part is not used.
 
-		There is one extra option:
+		There is two extra options:
```
Comment (k-takata) [inline runtime/doc/eval.txt]: is -> are

**Comment by ZyX-I** [comment]
I would say that I use `map <…>` far more often. But `:terminal` creates a buffer like `:new`, so what you selected is more consistent.

File: runtime/doc/terminal.txt [hunk 5]
```diff
@@ -76,7 +76,7 @@ See option 'termsize' for controlling the size of the terminal window.
 
 Syntax ~
 
-:ter[minal] [command]				*:ter* *:terminal*
+:ter[minal] {++autoclose} [command]		*:ter* *:terminal*
```
Comment (ZyX-I) [inline runtime/doc/terminal.txt]: I think this should use square brackes as it is optional.

**Pull request closed without merging**